### ISSUE TITLE
Feature: Add ONNX model parser for protobuf-based ML models

### ISSUE THREAD

**Issue by oscarvalenzuelab** [oscarvalenzuelab/semantic-copycat-binarysniffer#22]
## Summary
Implement an ONNX (Open Neural Network Exchange) model parser to extract and analyze components from ONNX format models, which use protobuf serialization.

## Motivation
ONNX is becoming the standard for model interoperability across frameworks. We need to:
- Identify framework origins (PyTorch, TensorFlow, etc.)
- Extract operator types and model architecture
- Detect embedded metadata and potential vulnerabilities
- Verify model integrity

### New Extractor
Create `binarysniffer/extractors/onnx_model.py`:
```python
class ONNXModelExtractor(BaseExtractor):
    def can_handle(self, file_path: Path) -> bool:
        # Check for .onnx extension
        # Verify ONNX protobuf magic bytes
    
    def extract(self, file_path: Path) -> ExtractedFeatures:
        # Parse ONNX protobuf structure
        # Extract graph nodes and operators
        # Identify framework signatures
        # Extract metadata and properties
```

### ONNX Structure Analysis
- Parse ModelProto structure
- Extract GraphProto with nodes and tensors
- Identify operator types (Conv, BatchNorm, ReLU, etc.)
- Extract opset version and producer information
- Parse metadata_props for additional information

### Component Detection
```python
ONNX_OPERATORS = {
    'ai.onnx': ['Conv', 'BatchNormalization', 'Relu', 'MaxPool'],
    'ai.onnx.ml': ['TreeEnsembleClassifier', 'LinearRegressor'],
    'com.microsoft': ['FusedConv', 'QuantizeLinear']
}

FRAMEWORK_SIGNATURES = {
    'pytorch': ['aten::', 'torch.', '_caffe2::'],
    'tensorflow': ['tf.', 'tensorflow::'],
    'sklearn': ['sklearn.', 'SklearnTransform']
}
```

## Implementation Steps
1. [ ] Create ONNXModelExtractor class
2. [ ] Implement protobuf parser (with optional onnx library)
3. [ ] Extract model graph structure
4. [ ] Identify framework-specific patterns
5. [ ] Add operator statistics and model metrics
6. [ ] Create tests with various ONNX models

## Dependencies
- Optional: `onnx` library for full parsing (fallback to protobuf patterns)
- Use existing protobuf signatures in database

## Testing Requirements
- Test with ONNX models from different frameworks
- Verify extraction without onnx library installed
- Test with corrupted/malformed ONNX files
- Performance test with large models (>100MB)

## Success Criteria
- Extracts model architecture information
- Identifies source framework correctly
- Works with and without onnx library
- Provides model complexity metrics
- Detects unusual patterns or modifications

## References
- [ONNX Specification](https://onnx.ai/onnx/intro/index.html)
- [ONNX Model Zoo](https://github.com/onnx/models)
- [ONNX Protobuf Schema](https://github.com/onnx/onnx/blob/main/onnx/onnx.proto)

## Priority
MEDIUM - Important for ML model analysis but less critical than Pickle security
